mindustry: unbreak

Mindavi opened the pull request:
###### Motivation for this change

This should fix:
- https://github.com/NixOS/nixpkgs/issues/128929 (but then for 21.11)
- https://github.com/NixOS/nixpkgs/issues/109798.

jdk15 is not supported anymore by oracle / openjdk, but since it's just a game, I don't expect that's a big issue in practice.

I _think_ propagating it like this is the right way to go about it, but please let me know if that seems wrong (and I'd like to hear the proper solution too then). I might have misunderstood how it works, but this seems like a good workaround to me.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Mindavi committed:
mindustry: unbreak by building with jdk15

Fixes: https://github.com/NixOS/nixpkgs/issues/128929

Mindavi committed:
mindustry: mitigate issue with missing glew

Fixes: https://github.com/NixOS/nixpkgs/issues/109798

legendofmiracles commented:
Result of `nixpkgs-review pr 146557` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>mindustry</li>
    <li>mindustry-wayland</li>
  </ul>
</details>

fgaz reviewed (approved):
Looks good and works for me!

SuperSandro2000 commented on pkgs/games/mindustry/default.nix:
```diff
@@ -155,11 +164,25 @@ stdenv.mkDerivation rec {
 
     pushd Mindustry
   '' + optionalString enableClient ''
+    glewlib=${glew.out}/lib/libGLEW.so
+    sdllib=${SDL2}/lib/libSDL2.so
+    if ! test -f "$glewlib"; then
+      echo "libGLEW not found"
+      exit 1
+    else
+      echo "libGLEW found"
+    fi
+    if ! test -f "$sdllib"; then
+      echo "libSDL2 not found"
+      exit 1
+    else
+      echo "libSDL2 found"
+    fi
```
We don't need any of the checking code if we use ``lib.getLib glew``.

Mindavi replied:
Thanks, that should be a nicer solution.

Mindavi replied:
`lib.getLib` only returns the path that contains the libraries, still not checking if the _wanted_ library is available.

Mindavi replied:
I did add nicer logging and added usage of the `lib.getLib` functionality. Still think it's nice to validate that the files actually exist.

SuperSandro2000 replied:
Why shouldn't the files exist? They are fundamental to the libraries. I doubt they get ever removed.

Mindavi replied:
I removed the checks.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/414

symphorien commented:
nix-review fails with 
```
error: anonymous function at /home/symphorien/.cache/nixpkgs-review/pr-146557/nixpkgs/pkgs/games/mindustry/default.nix:1:1 called with unexpected argument 'jdk'

       at /home/symphorien/.cache/nixpkgs-review/pr-146557/nixpkgs/lib/customisation.nix:69:16:

           68|     let
           69|       result = f origArgs;
             |                ^
           70|
(use '--show-trace' to show detailed location information)
```

Mindavi committed:
mindustry-server: use jdk15 to build server

Mindavi commented:
I think I've fixed this locally. It was due to this: https://github.com/NixOS/nixpkgs/pull/119444, where the jdk was changed to jdk11 (which doesn't work at all for mindustry). Will test the builds and fix it.

Mindavi committed:
mindustry: fix eval error due to #119444

The PR https://github.com/NixOS/nixpkgs/pull/119444 broke the build for mindustry
even further. This patch fixes that again so it evals and builds properly.

Mindavi committed:
mindustry: format mindustry in top-level/all-packages nicely

Mindavi commented:
Builds are not done yet, but this should be ok now.

SuperSandro2000 commented on pkgs/games/mindustry/default.nix:
```diff
@@ -157,9 +165,11 @@ stdenv.mkDerivation rec {
   '' + optionalString enableClient ''
     gradle --offline --no-daemon jnigenBuild -Pbuildversion=${buildVersion}
     gradle --offline --no-daemon sdlnatives -Pdynamic -Pbuildversion=${buildVersion}
+    glewlib=${lib.getLib glew}/lib/libGLEW.so
+    sdllib=${lib.getLib SDL2}/lib/libSDL2.so
```
```suggestion
```

SuperSandro2000 commented on pkgs/games/mindustry/default.nix:
```diff
@@ -157,9 +165,11 @@ stdenv.mkDerivation rec {
     patchelf ../Arc/backends/backend-sdl/libs/linux64/libsdl-arc*.so \
-      --add-needed ${glew.out}/lib/libGLEW.so \
-      --add-needed ${SDL2}/lib/libSDL2.so
+      --add-needed $glewlib \
+      --add-needed $sdllib
```
```suggestion
      --add-needed ${lib.getLib glew}/lib/libGLEW.so \
      --add-needed ${lib.getLib SDL2}/lib/libSDL2.so
```

fgaz commented:
Unfortunately I don't have much time to backport manually. Pinging people who might be interested: @maxhbr @Mindavi @petabyteboy
